tt_eager/tt_dnn/op_library/sharded/kernels/dataflow/reader_unary_sharded_blocks_interleaved_start_id.cpp

```diff
@@ -39,6 +50,22 @@ void kernel_main() {
             noc_async_read_barrier();
         }
         curr_tile_id += input_width_offset_tiles;
-    } 
-    cb_push_back(cb_id_in0, block_num_tiles);
+        if constexpr (convert_df && reserving_kernel) {
+            // push back early for data to be converted as soon as possible
+            cb_push_back(cb_id_in0, block_width_tiles);
+        }
```

Comment:
Doing a push here would make sense if we were barrier-ing here as well, but since we barrier in the inner loop I think it would make more sense to move the push back into the inner loop as well so that compute can get started on the available data immediately, unless we have some heuristics on how many tiles before we barrier and move it

Reply:
We're going to adjust how the barrier is called due to https://github.com/tenstorrent-metal/tt-metal/issues/6729 to not always be in the inner loop. However, it'll always be in the outer loop. See https://github.com/tenstorrent-metal/tt-metal/compare/pjosipovic/6729 for details.